app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/ProfileEditingActivity.java

```diff
@@ -48,28 +55,45 @@ protected void onCreate(Bundle savedInstanceState) {
         View view = binding.getRoot();
         setContentView(view);
         vmodel = new ViewModelProvider(this, new EditProfileViewModelFactory((NeighborFoodApplication) this.getApplication())).get(EditProfileViewModel.class);
-        vmodel.getCurrentUser().observe(this,user -> {
-            updateUserFields(user);
-        });
+        vmodel.getCurrentUser().observe(this, this::updateUserFields);
         setContentView(R.layout.activity_profile_editing);
         Toolbar toolbar = findViewById(R.id.profileEditToolbar);
         setSupportActionBar(toolbar);
-
         ppView = findViewById(R.id.profilePictureImageView);
         ppView.setOnClickListener(this::onClick);
-
+        findViewById(R.id.profileEditAddLinkButton).setOnClickListener(this::onClick);
+        linksLayout = (LinearLayout) findViewById(R.id.profileEditLinksLayout);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
     }
 
-    public void updateUserFields(User user) {
-        if(user != null){
-            ((TextView)findViewById(R.id.nameValue)).setText(user.getFirstName());
-            ((TextView)findViewById(R.id.surnameValue)).setText(user.getLastName());
-            ((TextView)findViewById(R.id.emailValue)).setText(user.getEmail());
-            Picasso.with(this).load(user.getProfilePictureURI()).fit().into(ppView);
+    private void updateUserFields(User user) {
+        if(user == null){
+            return;
         }
+        ((TextView)findViewById(R.id.nameValue)).setText(user.getFirstName());
+        ((TextView)findViewById(R.id.surnameValue)).setText(user.getLastName());
+        ((TextView)findViewById(R.id.emailValue)).setText(user.getEmail());
+        ((TextView)findViewById(R.id.bioValue)).setText(user.getBio());
+        Picasso.with(this).load(user.getProfilePictureURI()).fit().into(ppView);
+        updateUserLinks(user);
 
     }
 
+    private void updateUserLinks(User user){
```

Comment:
we should add more restrictions to user later on when adding links in order to have correct links format and I personally would have preferred to have the main three websites(for example facebook,twitter,instagram) to be filled since I think it's easier to manage,control the format and display them later on. But as we discussed before it's good this way too since it's more modular and general then the other alternative so this is just a general remark.

Reply:
Yes, as you said, I think it's more modular if we keep it as a simple link on the model side, but have the "known" social links to be differentiated on the view side, where we display the icon according to the type of the link, by looking at the domain name, instead of having distinct fields for each social media.
I think it's a good point to add more restrictions on the links and especially perform deeper checks on the content